packages/angular_devkit/build_angular/src/tools/esbuild/stylesheets/sass-language.ts

```diff
@@ -97,31 +103,49 @@ async function compileString(
             url,
             options: FileImporterWithRequestContextOptions,
           ): Promise<URL | null> => {
+            let resolved = resolutionCache.get(url);
+            if (resolved !== undefined) {
+              return resolved;
+            }
+
+            // Default to not found
+            resolved = null;
+
             const result = await resolveUrl(url, options);
             if (result.path) {
-              return pathToFileURL(result.path);
+              resolved = pathToFileURL(result.path);
+            } else {
+              // Check for package deep imports
+              const parts = url.split('/');
+              const hasScope = parts.length >= 2 && parts[0].startsWith('@');
+              const [nameOrScope, nameOrFirstPath, ...pathPart] = parts;
+              const packageName = hasScope ? `${nameOrScope}/${nameOrFirstPath}` : nameOrScope;
+
+              // Caching package root locations is particularly beneficial for `@material/*` packages
+              // which extensively use deep imports.
+              let packageRoot = packageRootCache.get(packageName);
+              if (packageRoot === undefined) {
+                const packageResult = await resolveUrl(packageName + '/package.json', options);
+                if (packageResult.path) {
+                  packageRoot = dirname(packageResult.path);
+                  packageRootCache.set(packageName, packageRoot);
+                }
+              }
+
+              if (packageRoot) {
+                resolved = pathToFileURL(
+                  join(
+                    packageRoot,
+                    !hasScope && nameOrFirstPath ? nameOrFirstPath : '',
```

Comment:
Question: Doesn't this drop the `@angular` part of the package name? Maybe I'm missing something, but I feel like it would generate `path/to/node_modules/material/button/button.scss`?

Reply:
In the scoped case of `@angular/material/button` (assuming an actual deep import and not an exports entry):
`packageRoot`: `path/to/node_modules/@angular/material`
`hasScope`: `true`
`nameOrFirstPath`: unused but contains `material`
`pathPart`: `['button']`
which becomes `join('path/to/node_modules/@angular/material', '', 'button')`